### DISCUSSION TITLE
Multi value parameter

### DISCUSSION THREAD

**Discussion by odeddror** [DaxStudio/DaxStudio#827] [General]
Hi there,
Please see my question below
I've tried to pass Red,Blue color but its a string parameter I got no record 
But from some reason you can do it with Query Builder
Thanks,
Oded Dror

/* START QUERY BUILDER */
EVALUATE
SUMMARIZECOLUMNS(
    DimVehicle[Make],
    DimVehicle[Model],
    DimVehicle[Color],
    KEEPFILTERS( TREATAS( {"Red","Blue"}, DimVehicle[Color] )),
    "Gross Profit", [Gross Profit]
)
ORDER BY 
    DimVehicle[Make] ASC,
    DimVehicle[Model] ASC,
    DimVehicle[Color] ASC
/* END QUERY BUILDER */

// How to run the query below and pass multi values parameters like the query builder query above?

EVALUATE
SUMMARIZECOLUMNS(
 DimCLient[ClientName]
,DimVehicle[Make]
,DimVehicle[Model]
,DimVehicle[Color]
,FILTER(
VALUES(DimVehicle[Color]),DimVehicle[Color] In { @Colour })
,"Sales", SUM(FactSales[SalePrice])
)

Thanks

**Comment by dgosbell** [comment]
Unfortunately this is a limitation of the DAX language, not something I can fix in DAX Studio. DAX only supports single value parameters, it does not have the ability to pass multiple values.

So when you have a statement like the following `VALUES(DimVehicle[Color]),DimVehicle[Color] In { @colour })` if you pass in `Red,Blue`it actually gets expanded as `VALUES(DimVehicle[Color]),DimVehicle[Color] In { "Red,Blue" })`

If you are using these multi value parameters from a paginated report you could the RSCustomDaxFilter function (see https://blog.crossjoin.co.uk/2019/11/03/power-bi-report-builder-and-rscustomdaxfilter/) but this is not a DAX function, it is actually part of the paginated report processing so queries with this function will not work in DAX Studio. 

The other option use a DAX based approach like this https://blog.crossjoin.co.uk/2018/05/16/a-new-approach-to-handling-ssrs-multi-valued-parameters-in-dax-queries/

**Reply by odeddror** [reply]
I was reading an article by Adam Aspin and thought if it possible to pass an array as parameter in DAX.
Thanks for the clarification
Oded Dror
